### ISSUE TITLE
Some demonet vchains are down because of repeated disk errors

### ISSUE THREAD

**Issue by netoneko** [orbs-network/boyarin#100]
```
ID                          NAME                                   IMAGE                                                                                                          NODE                DESIRED STATE       CURRENT STATE                 ERROR                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          PORTS
04ri846sczf3vkh4sdqfttv0x   orbs-network-chain-1003-stack.1        orbsnetwork/node:experimental-js@sha256:cc22b8dfa43a21282b5894b09f37ec742f59568ec851171f7a3dd11097eb5da3       ip-172-31-4-76      Running             Starting 18 seconds ago                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
ze4zy6mvcp9mjt1b9c9iw9x2b   orbs-network-chain-1000-stack.1        orbsnetwork/node:experimental-extlib@sha256:e777e2b7826c8e045d4eeb0b6f1f521f34fb5bd1c0a605b903bea32180f7a959   ip-172-31-4-76      Running             Starting about a minute ago                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
d7gyw97219o7q9dewni7on89j   orbs-network-chain-1003-stack.1        orbsnetwork/node:experimental-js@sha256:cc22b8dfa43a21282b5894b09f37ec742f59568ec851171f7a3dd11097eb5da3       ip-172-31-4-76      Shutdown            Failed 29 seconds ago         "starting container failed: OCI runtime create failed: container_linux.go:346: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/mnt/data/var/lib/docker/plugins/bacfb66c29de7b3de1ec20da1403ab6d131c0a5d5376ec1f6df6dd4dd41fc660/propagated-mount/volumes/a328846cd5b4979d68a8c58a9bdfeee657b34de7-1003-logs/data\\\" to rootfs \\\"/mnt/data/var/lib/docker/overlay2/8008c3ded78f1981e17151e5d8fcedff0010267424b7890377bbd4dad7052f33/merged\\\" at \\\"/opt/orbs/logs\\\" caused \\\"stat /mnt/data/var/lib/docker/plugins/bacfb66c29de7b3de1ec20da1403ab6d131c0a5d5376ec1f6df6dd4dd41fc660/propagated-mount/volumes/a328846cd5b4979d68a8c58a9bdfeee657b34de7-1003-logs/data: no such file or directory\\\"\"": unknown"   
8a4sqd8f8g1ctjpz2ml1cc9sh   orbs-network-chain-1000-stack.1        orbsnetwork/node:experimental-extlib@sha256:e777e2b7826c8e045d4eeb0b6f1f521f34fb5bd1c0a605b903bea32180f7a959   ip-172-31-4-76      Shutdown            Failed about a minute ago     "starting container failed: OCI runtime create failed: container_linux.go:346: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/mnt/data/var/lib/docker/plugins/bacfb66c29de7b3de1ec20da1403ab6d131c0a5d5376ec1f6df6dd4dd41fc660/propagated-mount/volumes/a328846cd5b4979d68a8c58a9bdfeee657b34de7-1000-logs/data\\\" to rootfs \\\"/mnt/data/var/lib/docker/overlay2/7a7c437806d4abbd9d42134fa9bdeba015c0af3ab082ad33f4118302e4034cf4/merged\\\" at \\\"/opt/orbs/logs\\\" caused \\\"stat /mnt/data/var/lib/docker/plugins/bacfb66c29de7b3de1ec20da1403ab6d131c0a5d5376ec1f6df6dd4dd41fc660/propagated-mount/volumes/a328846cd5b4979d68a8c58a9bdfeee657b34de7-1000-logs/data: no such file or directory\\\"\"": unknown"   
tyi18xyve40sjuethh26940is   orbs-network-chain-1003-stack.1        orbsnetwork/node:experimental-js@sha256:cc22b8dfa43a21282b5894b09f37ec742f59568ec851171f7a3dd11097eb5da3       ip-172-31-4-76      Shutdown            Failed about a minute ago     "starting container failed: OCI runtime create failed: container_linux.go:346: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/mnt/data/var/lib/docker/plugins/bacfb66c29de7b3de1ec20da1403ab6d131c0a5d5376ec1f6df6dd4dd41fc660/propagated-mount/volumes/a328846cd5b4979d68a8c58a9bdfeee657b34de7-1003-logs/data\\\" to rootfs \\\"/mnt/data/var/lib/docker/overlay2/b5f6f64111b53a18d82a2337c8b7f6f924986b690bf7e3f70be7dd2868ff36ec/merged\\\" at \\\"/opt/orbs/logs\\\" caused \\\"stat /mnt/data/var/lib/docker/plugins/bacfb66c29de7b3de1ec20da1403ab6d131c0a5d5376ec1f6df6dd4dd41fc660/propagated-mount/volumes/a328846cd5b4979d68a8c58a9bdfeee657b34de7-1003-logs/data: no such file or directory\\\"\"": unknown"   
s2cifkyc812j8vfxcgyv1k8b6   orbs-network-chain-1000-stack.1        orbsnetwork/node:experimental-extlib@sha256:e777e2b7826c8e045d4eeb0b6f1f521f34fb5bd1c0a605b903bea32180f7a959   ip-172-31-4-76      Shutdown            Failed 2 minutes ago          "starting container failed: OCI runtime create failed: container_linux.go:346: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/mnt/data/var/lib/docker/plugins/bacfb66c29de7b3de1ec20da1403ab6d131c0a5d5376ec1f6df6dd4dd41fc660/propagated-mount/volumes/a328846cd5b4979d68a8c58a9bdfeee657b34de7-1000-logs/data\\\" to rootfs \\\"/mnt/data/var/lib/docker/overlay2/eede5320feacd3324eaffd2eeffe0fc4b58e9f6b0e1e11997045bc4c9ab7fe4b/merged\\\" at \\\"/opt/orbs/logs\\\" caused \\\"stat /mnt/data/var/lib/docker/plugins/bacfb66c29de7b3de1ec20da1403ab6d131c0a5d5376ec1f6df6dd4dd41fc660/propagated-mount/volumes/a328846cd5b4979d68a8c58a9bdfeee657b34de7-1000-logs/data: no such file or directory\\\"\"": unknown"   
0durm4th9vvrb1rep225ky7d8   orbs-network-chain-1003-stack.1        orbsnetwork/node:experimental-js@sha256:cc22b8dfa43a21282b5894b09f37ec742f59568ec851171f7a3dd11097eb5da3       ip-172-31-4-76      Shutdown            Failed 3 minutes ago          "starting container failed: OCI runtime create failed: container_linux.go:346: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/mnt/data/var/lib/docker/plugins/bacfb66c29de7b3de1ec20da1403ab6d131c0a5d5376ec1f6df6dd4dd41fc660/propagated-mount/volumes/a328846cd5b4979d68a8c58a9bdfeee657b34de7-1003-logs/data\\\" to rootfs \\\"/mnt/data/var/lib/docker/overlay2/4709868a97d07e52c2c3908f88d8fab9798a83a23f1c2fecc92145803a529ea2/merged\\\" at \\\"/opt/orbs/logs\\\" caused \\\"stat /mnt/data/var/lib/docker/plugins/bacfb66c29de7b3de1ec20da1403ab6d131c0a5d5376ec1f6df6dd4dd41fc660/propagated-mount/volumes/a328846cd5b4979d68a8c58a9bdfeee657b34de7-1003-logs/data: no such file or directory\\\"\"": unknown"   
fwhx1wak09n8qb7njf7xwzf18   orbs-network-chain-1000-stack.1        orbsnetwork/node:experimental-extlib@sha256:e777e2b7826c8e045d4eeb0b6f1f521f34fb5bd1c0a605b903bea32180f7a959   ip-172-31-4-76      Shutdown            Failed 4 minutes ago          "starting container failed: OCI runtime create failed: container_linux.go:346: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/mnt/data/var/lib/docker/plugins/bacfb66c29de7b3de1ec20da1403ab6d131c0a5d5376ec1f6df6dd4dd41fc660/propagated-mount/volumes/a328846cd5b4979d68a8c58a9bdfeee657b34de7-1000-logs/data\\\" to rootfs \\\"/mnt/data/var/lib/docker/overlay2/a2ff8cb491ed150b8a0598e8b844a784b87507fcb6992b84fe8e8b4dd4df7ffa/merged\\\" at \\\"/opt/orbs/logs\\\" caused \\\"stat /mnt/data/var/lib/docker/plugins/bacfb66c29de7b3de1ec20da1403ab6d131c0a5d5376ec1f6df6dd4dd41fc660/propagated-mount/volumes/a328846cd5b4979d68a8c58a9bdfeee657b34de7-1000-logs/data: no such file or directory\\\"\"": unknown"   
zph7v0pxh75olhuyg3lxohcgl   orbs-network-chain-1003-stack.1        orbsnetwork/node:experimental-js@sha256:cc22b8dfa43a21282b5894b09f37ec742f59568ec851171f7a3dd11097eb5da3       ip-172-31-4-76      Shutdown            Failed 4 minutes ago          "starting container failed: OCI runtime create failed: container_linux.go:346: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/mnt/data/var/lib/docker/plugins/bacfb66c29de7b3de1ec20da1403ab6d131c0a5d5376ec1f6df6dd4dd41fc660/propagated-mount/volumes/a328846cd5b4979d68a8c58a9bdfeee657b34de7-1003-logs/data\\\" to rootfs \\\"/mnt/data/var/lib/docker/overlay2/b497a8c61131954b65fd81643ea1ccae65922546ddc31aaac1e1ca6da58b2861/merged\\\" at \\\"/opt/orbs/logs\\\" caused \\\"stat /mnt/data/var/lib/docker/plugins/bacfb66c29de7b3de1ec20da1403ab6d131c0a5d5376ec1f6df6dd4dd41fc660/propagated-mount/volumes/a328846cd5b4979d68a8c58a9bdfeee657b34de7-1003-logs/data: no such file or directory\\\"\"": unknown"   
sabjmywqtd3rotxd8bu7wqv37   orbs-network-chain-1000-stack.1        orbsnetwork/node:experimental-extlib@sha256:e777e2b7826c8e045d4eeb0b6f1f521f34fb5bd1c0a605b903bea32180f7a959   ip-172-31-4-76      Shutdown            Failed 5 minutes ago          "starting container failed: OCI runtime create failed: container_linux.go:346: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/mnt/data/var/lib/docker/plugins/bacfb66c29de7b3de1ec20da1403ab6d131c0a5d5376ec1f6df6dd4dd41fc660/propagated-mount/volumes/a328846cd5b4979d68a8c58a9bdfeee657b34de7-1000-logs/data\\\" to rootfs \\\"/mnt/data/var/lib/docker/overlay2/40191f9b1a8afef06162b18638d87056c12c7bc62c740c854396f8ce578ad9aa/merged\\\" at \\\"/opt/orbs/logs\\\" caused \\\"stat /mnt/data/var/lib/docker/plugins/bacfb66c29de7b3de1ec20da1403ab6d131c0a5d5376ec1f6df6dd4dd41fc660/propagated-mount/volumes/a328846cd5b4979d68a8c58a9bdfeee657b34de7-1000-logs/data: no such file or directory\\\"\"": unknown"   
r7muqjjmsqnqwkvcrmki08cgc   http-api-reverse-proxy-stack.1         nginx:latest@sha256:aeded0f2a861747f43a01cf1018cf9efe2bdd02afd57d2b11fcc7fcadc16ccd1                           ip-172-31-4-76      Running             Running 57 minutes ago                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       
im4m98jdha38bnk4zmyd66c4v   orbs-node-signer-service-stack.1       orbsnetwork/signer:experimental@sha256:7edfabf21bfc7b3f283e58dc6c7f7af59debba299d43519cb8b1c9c0ada63063        ip-172-31-4-76      Running             Running 57 minutes ago                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       
p77uey14786ns7kunrunjcxhl    \_ orbs-node-signer-service-stack.1   orbsnetwork/signer:experimental@sha256:7edfabf21bfc7b3f283e58dc6c7f7af59debba299d43519cb8b1c9c0ada63063        ip-172-31-4-76      Shutdown            Failed 57 minutes ago         "task: non-zero exit (2)"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
ro3d7ukp94f27uyo8r3nagu18   http-api-reverse-proxy-stack.1         nginx:latest@sha256:aeded0f2a861747f43a01cf1018cf9efe2bdd02afd57d2b11fcc7fcadc16ccd1                           ip-172-31-4-76      Shutdown            Complete 57 minutes ago                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
```

**Comment by amir-arad** [comment]
At some point the node crashes due to disks that become unavailable. When the disk becomes available again it fails to mount, and so Swarm tries to restart the service but fails.
Maybe killing and restarting the service externally will solve this?

**Comment by amir-arad** [comment]
reproduce by running the following command in the node manager:

`docker service ps --no-trunc $(docker service ls -q)`

**Comment by netoneko** [comment]
Additional logs from an affected machine (recovered via `journalctl -u docker.service --no-page`):

```
Nov 26 19:30:31 ip-172-31-12-76 dockerd[3319]: time="2019-11-26T19:30:31Z" level=error msg="time=\"2019-11-26T19:30:30Z\" level=warning msg=\"device does not match\" deviceName=xvda deviceRX=^xvd[f-p]$ host=\"unix:///var/run/rexray/621905625.sock\" instanceID=\"ebs=i-07c403227ca0d4dce,availabilityZone=eu-central-1a&region=eu-central-1\" integrationDriver=linux osDriver=linux server=platinum-gull-ke service=ebs storageDriver=libstorage time=1574796630839 txCR=1574796630 txID=cf28a3e4-1d99-49ba-77a3-3afaca169406 " plugin=341520add81edd6423d5c8ad5a9b28c74f2a51b5baa7a0a4636d6a8689528e39
Nov 26 19:30:31 ip-172-31-12-76 dockerd[3319]: time="2019-11-26T19:30:31Z" level=error msg="time=\"2019-11-26T19:30:30Z\" level=warning msg=\"device does not match\" deviceName=xvda1 deviceRX=^xvd[f-p]$ host=\"unix:///var/run/rexray/621905625.sock\" instanceID=\"ebs=i-07c403227ca0d4dce,availabilityZone=eu-central-1a&region=eu-central-1\" integrationDriver=linux osDriver=linux server=platinum-gull-ke service=ebs storageDriver=libstorage time=1574796630839 txCR=1574796630 txID=cf28a3e4-1d99-49ba-77a3-3afaca169406 " plugin=341520add81edd6423d5c8ad5a9b28c74f2a51b5baa7a0a4636d6a8689528e39
```

This is accompanied by aggressive CPU hogging:

<img width="1434" alt="Screen Shot 2019-11-26 at 21 39 13" src="https://user-images.githubusercontent.com/614125/69666612-34c94a80-1095-11ea-85c7-85ecdecf67d5.png">

**Comment by netoneko** [comment]
Noticed following error on demonet that possibly result in descheduling all vchains because of Docker daemon memory leak (node count 1):

```
Jan 11 21:03:02 ip-172-31-7-167 dockerd[1105]: time="2020-01-11T21:03:02.869105709Z" level=warning msg="underweighting node hmuxozmw6yyvdet3venzyno6a for service 8nbmtbez547i4y1iww82p1ln4 because it experienced 5 failures or rejections within 5m0s" module=node node.id=hmuxozmw6yyvdet3venzyno6a
Jan 11 21:05:41 ip-172-31-7-167 dockerd[1105]: time="2020-01-11T21:05:41.786030791Z" level=warning msg="underweighting node hmuxozmw6yyvdet3venzyno6a for service 3jwq0k0pc7i7vqyo92fgcx8j2 because it experienced 5 failures or rejections within 5m0s" module=node node.id=hmuxozmw6yyvdet3venzyno6a
Jan 11 21:05:41 ip-172-31-7-167 dockerd[1105]: time="2020-01-11T21:05:41.887780311Z" level=warning msg="underweighting node hmuxozmw6yyvdet3venzyno6a for service dpaffa75a14ohbx8tijtfu9bj because it experienced 5 failures or rejections within 5m0s" module=node node.id=hmuxozmw6yyvdet3venzyno6a
Jan 11 21:10:32 ip-172-31-7-167 dockerd[15564]: time="2020-01-11T21:10:32.630580096Z" level=warning msg="underweighting node hmuxozmw6yyvdet3venzyno6a for service scle50bfbimxwpmmnwh1czxto because it experienced 5 failures or rejections within 5m0s" module=node node.id=hmuxozmw6yyvdet3venzyno6a
Jan 11 21:11:20 ip-172-31-7-167 dockerd[15564]: time="2020-01-11T21:11:20.059294928Z" level=warning msg="underweighting node hmuxozmw6yyvdet3venzyno6a for service 3jwq0k0pc7i7vqyo92fgcx8j2 because it experienced 5 failures or rejections within 5m0s" module=node node.id=hmuxozmw6yyvdet3venzyno6a
Jan 11 21:13:02 ip-172-31-7-167 dockerd[15564]: time="2020-01-11T21:13:02.712378232Z" level=warning msg="underweighting node n8undeuxtj9hdfymkml8deous for service 8nbmtbez547i4y1iww82p1ln4 because it experienced 5 failures or rejections within 5m0s" module=node node.id=hmuxozmw6yyvdet3venzyno6a
```

**Comment by netoneko** [comment]
Resolved by #138
